File: Easy/25runLineEncoding.py
def runLengthEncoding(string):
    encodedStringCharacters = []
    currentRunLength = 1

    for i in range(1, len(string)):
        currentCharacter = string[i]
        previousCharacter = string[i-1]

        if currentCharacter != previousCharacter or currentRunLength == 9:
            encodedStringCharacters.append(str(currentRunLength))
            encodedStringCharacters.append(previousCharacter)
            currentRunLength = 0
        currentRunLength += 1

    encodedStringCharacters.append(str(currentRunLength))
    encodedStringCharacters.append(string[len(string)-1])
    return ''.join(encodedStringCharacters)


print(runLengthEncoding('aaaaaaaaaaaaabbbccddd'))

File: Easy/25runLineEncoding.js
function runLengthEncoding(string) {
  let encodedStringCharacter = [];
  let currentRunLength = 1;
  for (let i = 1; i < string.length; i++) {
    currentCharacter = string[i];
    previousCharacter = string[i - 1];
    if (previousCharacter != currentCharacter || currentRunLength == 9) {
      encodedStringCharacter.push(currentRunLength.toString());
      encodedStringCharacter.push(previousCharacter);
      currentRunLength = 0;
    }
    currentRunLength++;
  }
  encodedStringCharacter.push(currentRunLength.toString());
  encodedStringCharacter.push(string[string.length - 1]);
  return encodedStringCharacter.join("");
}

console.log(runLengthEncoding("aaaaaaaaaaaaabbbccddd"));
